### PR TITLE
Add negative test for numatune

### PR DESCRIPTION
**Pull request by kylazhang**
Signed-off-by: Kylazhang <weizhan@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py [hunk 1]
```diff
@@ -255,6 +257,10 @@ def run(test, params, env):
                 get_numa_parameter(test, params, cgstop)
             else:
                 set_numa_parameter(test, params, cgstop)
+
+            aft_pid_libvirtd = process.getoutput("pidof libvirtd")
+            if ori_pid_libvirtd != aft_pid_libvirtd:
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py]: checking libvirtd is running also helps in asserting, as sometime libvirtd would stopped running.
```python
if not utils_libvirtd.libvirtd_is_running()  or ori_pid_libvirtd != aft_pid_libvirtd:
```

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_numatune.cfg [hunk 2]
```diff
@@ -147,6 +147,13 @@
                                             numa_nodeset = 0-2,,^1
                                         - single:
                                             numa_nodeset = x
+                                        - large_num:
+                                            numa_nodeset = 10000000000
+                                        - minus_num:
+                                            numa_nodeset = -1
+                                        - exceed_num:
+                                            no config
+                                            numa_nodeset = 11
```
Comment (balamuruhans) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_numatune.cfg]: here we assume that host cannot have numanode 11, let's check for the host numa nodes existing and then set it max available + 1.

File: libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py (hunk 1 above)
Comment (kylazhang) [reply libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py]: updated

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_numatune.cfg (hunk 2 above)
Comment (kylazhang) [reply libvirt/tests/cfg/virsh_cmd/domain/virsh_numatune.cfg]: updated

File: libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py [hunk 3]
```diff
@@ -106,12 +107,18 @@ def set_numa_parameter(test, params, cgstop):
...
+    exceed_num = "yes" == params.get("exceed_num", "no")
 
     # Get host numa node list
     host_numa_node = utils_misc.NumaInfo()
     node_list = host_numa_node.online_nodes_withmem
     logging.debug("host online nodes with memory %s", node_list)
 
+    # Get host numa node number if exceed_num
+    if exceed_num:
+        nodeset = str(len(host_numa_node.all_nodes)+1)
```
Comment (balamuruhans) [inline libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py]: space around the operator `+`,

**Comment by kylazhang** [comment]
```
(.libvirt-ci-venv-ci-runtest-AjlfNu) [root@libvirt-rhel-8 tp-libvirt]# avocado run --vt-type libvirt virsh.numatune.negative_testing.set_numa_parameter.running_guest.change_nodeset.options.live.exceed_num
JOB ID     : 47183d37a47b43bddccce2afa9ecea5186a74f1a
JOB LOG    : /root/avocado/job-results/job-2020-02-09T22.46-47183d3/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.numatune.negative_testing.set_numa_parameter.running_guest.change_nodeset.options.live.exceed_num: PASS (8.43 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 9.99 s
(.libvirt-ci-venv-ci-runtest-AjlfNu) [root@libvirt-rhel-8 tp-libvirt]# vim /root/avocado/job-results/job-2020-02-09T22.46-47183d3/job.log
(.libvirt-ci-venv-ci-runtest-AjlfNu) [root@libvirt-rhel-8 tp-libvirt]# avocado run --vt-type libvirt virsh.numatune.negative_testing.set_numa_parameter.running_guest.change_nodeset.options.live.large_num
JOB ID     : 78dae3171ad89aea4ccf2af66da1945104fc7c1a
JOB LOG    : /root/avocado/job-results/job-2020-02-09T22.47-78dae31/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.numatune.negative_testing.set_numa_parameter.running_guest.change_nodeset.options.live.large_num: PASS (6.21 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 7.75 s
(.libvirt-ci-venv-ci-runtest-AjlfNu) [root@libvirt-rhel-8 tp-libvirt]# avocado run --vt-type libvirt virsh.numatune.negative_testing.set_numa_parameter.running_guest.change_nodeset.options.live.minus_num
JOB ID     : d6aad023f159fc0986bd72dd3f542c22356da29f
JOB LOG    : /root/avocado/job-results/job-2020-02-09T22.48-d6aad02/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.numatune.negative_testing.set_numa_parameter.running_guest.change_nodeset.options.live.minus_num: PASS (6.29 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 7.88 s
```

File: libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py (hunk 3 above)
Comment (kylazhang) [reply libvirt/tests/src/virsh_cmd/domain/virsh_numatune.py]: updated, thanks

**Comment by kylazhang** [comment]
Merge it according to comments and approval.

**Pull request merged**